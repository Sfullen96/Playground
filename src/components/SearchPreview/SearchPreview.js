import React from 'react';
import SearchPreviewItem from '../SearchPreviewItem/SearchPreviewItem';
import './SearchPreview.scss';

/*
* SearchPreview component
* Renders the drop-down of preview
* items based on a users input
*
* @param {Any} props - Holds the array of search previews
* */
const SearchPreview = (props) => {
    const {
        searchResults,
        onClick,
        onMouseOut,
        onMouseOver,
    } = props;

    if (searchResults && !searchResults.length) {
        return (
            <h1>
                Loading...
            </h1>
        );
    }

    return (
        // eslint-disable-next-line jsx-a11y/mouse-events-have-key-events
        <div
            className="search-preview"
            onMouseOut={onMouseOut}
            onMouseOver={onMouseOver}
        >
            <ul>
                {
                    searchResults
                    && searchResults
                        .map(result => (
                            <SearchPreviewItem
                                key={result.bookingId}
                                searchPreviewItem={result}
                                onClick={onClick}
                            />
                        ))
                }
            </ul>
        </div>
    );
};

export default SearchPreview;
