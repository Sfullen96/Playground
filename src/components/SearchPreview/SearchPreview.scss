@import "../../css/mixins";

@include i-phone-retina() {
    .search-preview {
        position: absolute;
        left: 0;
        right: 0;
        margin: 0 auto;
        width: 100%;
        top: 68px;
        z-index: 9999;
        border: 1px solid #e6e6e6;
        height: 0;
        ul {
            list-style-type: none;
            background-color: #fff;
            padding-left: 0;
        }
    }
}
