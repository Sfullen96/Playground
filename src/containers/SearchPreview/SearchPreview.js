import React, { Component } from 'react';
import SearchPreviewItem from '../../components/SearchPreviewItem/SearchPreviewItem';
import './SearchPreview.scss';

/*
* SearchPreview component
* Renders the drop-down of preview
* items based on a users input
*
* @param {Any} props - Holds the array of search previews
* */
class SearchPreview extends Component {
    // eslint-disable-next-line consistent-return
    componentDidMount() {
        const { searchResults } = this.props;

        if (!searchResults.length) {
            return (
                <h1>
                    Loading...
                </h1>
            );
        }
    }

    render() {
        const {
            searchResults,
            onClick,
            onMouseOut,
            onMouseOver,
        } = this.props;
        return (
            // eslint-disable-next-line jsx-a11y/mouse-events-have-key-events
            <div
                className="search-preview"
                onMouseOut={onMouseOut}
                onMouseOver={onMouseOver}
            >
                <ul>
                    {
                        searchResults
                            .map(result => (
                                <SearchPreviewItem
                                    key={result.bookingId}
                                    searchPreviewItem={result}
                                    onClick={onClick}
                                />
                            ))
                    }
                </ul>
            </div>
        );
    }
}

export default SearchPreview;
