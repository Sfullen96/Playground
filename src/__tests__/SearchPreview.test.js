/* eslint-disable no-undef */
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SearchPreview from '../components/SearchPreview/SearchPreview';

Enzyme.configure({ adapter: new Adapter() });

/*
* Test that the SearchPreview renders
* and matches it's latest snapshot
* */
it('Should render the search preview and it should match the snapshot', () => {
    const component = shallow(<SearchPreview />);
    expect(component).toMatchSnapshot();
});
