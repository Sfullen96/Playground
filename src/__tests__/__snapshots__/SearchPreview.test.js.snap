// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the search preview and it should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchPreview />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul />,
      "className": "search-preview",
      "onMouseOut": undefined,
      "onMouseOver": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul />,
        "className": "search-preview",
        "onMouseOut": undefined,
        "onMouseOver": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
